### DISCUSSION TITLE
No Record was found for a nested update on relation

### DISCUSSION THREAD

**Discussion by saeed-latifi** [prisma/prisma#19656] [Q&A]
### Bug description

hi
it is a course and has many lessons (or none) 

when update course category i want update thier lesson category too. if course has any lesson, it works. but if no lesson for course yet, and update course category it throw : "No Record was found for a nested update on relation Error"

and some info of course and lesson like title and category stored in more general "content" table

```
return await prismaProvider.course.update({
	where: { id: courseId },
	data: {
		content: { update: { description, title, categoryId, keywords: prismaKeywordUpdateHandler({ authorId, keywords }) } },
		lessons: { update: { data: { content: { update: { categoryId } } }, where: { courseId } } },	},
	select: coursePanelSelect(),
});
```

### Environment & setup

- OS: windows 10
- Database: postgre
- Node.js version: 18
- Prisma Version: 14

```

```

**Comment by ludralph** [comment]
Hi @VivoCyber 👋 

The error message suggests that there is no existing `lesson` record associated with the `Course`. You might want to ensure that a `lesson` record exists for `Course` before updating it. You can as well share the schema so we can have a clearer picture of how you modeled the relationship.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nurul3101** [reply]
@VivoCyber Did you had a chance to look at Raphael's last message?

Please let us know in case you still need any assistance.

**Reply by nurul3101** [reply]
@VivoCyber Please let us know in case you still need any assistance.  If the issue is resolved, please feel free to mark the discussion as answered.

**Reply by saeed-latifi** [reply]
thanks.
i changed this data model but its possible face same problem in other where.
so how can "ensure that a lesson record exists for Course before updating it" in update query. so many thanks if has a solotion

**Reply by saeed-latifi** [reply]
and its not usefull if check this existance in other query befor this one!

**Reply by LexQzim** [reply]
I'm facing the same issue and I wondering if it is possible to just skip the update if it is not available. Is there a nice solution with `undefined`?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
